pandas/tests/series/test_constructors.py

```diff
@@ -1810,6 +1810,18 @@ def test_constructor_bool_dtype_missing_values(self):
         expected = Series(True, index=[0], dtype="bool")
         tm.assert_series_equal(result, expected)
 
+    def test_constructor_int64_dtype(self):
+        # GH-44923
+        result = Series(["0", "1", "2"], dtype="int64")
```

Comment:
can you come up with a test case that goes through 2099-2100 but that _shouldn't_ get cast?  maybe ["0", "1", "1.1"]?

Reply:
Series(["0", "1", 0, 1], dtype="int64") this raises the specified ```ValueError```

Reply:
But shouldn't this get cast to ```int64``` as all can be type casted.

Reply:
Should I add the above example and change the code to fit it?

Reply:
@jbrockmendel there is no such case, so it is good to keep that line as it keeps the base covered.